### PR TITLE
Added modernize-use-auto to clang-tidy-checks

### PR DESCRIPTION
**Pull request by gnawme**
When `clang-tidy` checks were added to the Github workflow in #4636, `modernize-use-auto` was omitted.

Using `auto` can greatly enhance the readability and usability of the code.

### CODE REVIEW

**Comment by mvieth** [comment]
Personally, I would e.g. prefer `float` and `double` over `auto`, because they are still very short but more expressive. How about increasing MinTypeNameLength (https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html#cmdoption-arg-mintypenamelength) to 7? And maybe some of the for-loops with iterators can be converted to range-for-loops?

**Comment by gnawme** [comment]
> Personally, I would e.g. prefer `float` and `double` over `auto`, because they are still very short but more expressive. How about increasing MinTypeNameLength (https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html#cmdoption-arg-mintypenamelength) to 7? And maybe some of the for-loops with iterators can be converted to range-for-loops?

Thanks, yes, I think I'll follow both suggestions. Here is Part I, with MinTypeNameLength set

File: io/src/image_depth.cpp [hunk 1]
```diff
@@ -182,7 +182,7 @@ pcl::io::DepthImage::fillDepthImageRaw (unsigned width, unsigned height, unsigne
   short bad_point = std::numeric_limits<short>::quiet_NaN ();
   unsigned depthIdx = 0;
 
-  const unsigned short* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
+  auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```
Comment (larshg) [inline io/src/image_depth.cpp:185]: Can still be const here?

Comment (gnawme) [reply io/src/image_depth.cpp:185]: Yeah, that's an odd one.

File: io/src/image_depth.cpp [hunk 2]
```diff
@@ -229,7 +229,7 @@ pcl::io::DepthImage::fillDepthImage (unsigned width, unsigned height, float* dep
   float bad_point = std::numeric_limits<float>::quiet_NaN ();
   unsigned depthIdx = 0;
 
-  const unsigned short* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
+  auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```
Comment (larshg) [inline io/src/image_depth.cpp:232]:
```suggestion
  const auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```

File: io/src/image_depth.cpp [hunk 3]
```diff
@@ -275,7 +275,7 @@ pcl::io::DepthImage::fillDisparityImage (unsigned width, unsigned height, float*
   // focal length is for the native image resolution -> focal_length = focal_length_ / xStep;
   float constant = focal_length_ * baseline_ * 1000.0f / static_cast<float> (xStep);
 
-  const unsigned short* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
+  auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```
Comment (larshg) [inline io/src/image_depth.cpp:278]:
```suggestion
  const auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```

File: io/src/image_ir.cpp [hunk 4]
```diff
@@ -138,7 +138,7 @@ void pcl::io::IRImage::fillRaw (unsigned width, unsigned height, unsigned short*
 
   unsigned irIdx = 0;
 
-  const unsigned short* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
+  auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```
Comment (larshg) [inline io/src/image_ir.cpp:141]:
```suggestion
  const auto* inputBuffer = static_cast<const unsigned short*> (wrapper_->getData ());
```

**Comment by mvieth** [comment]
This pull request has gotten very big, far too big for a proper (and fast) review. Can you split it? Maybe move all commits after [Fixed merge error](https://github.com/PointCloudLibrary/pcl/pull/5397/commits/9db6178233d3c6ebdd8ac1623defaba8359bce73) (the first commit where all CI checks passed) to one or more additional pull requests? Thanks!

**Comment by gnawme** [comment]
> This pull request has gotten very big, far too big for a proper (and fast) review. Can you split it? Maybe move all commits after [Fixed merge error](https://github.com/PointCloudLibrary/pcl/pull/5397/commits/9db6178233d3c6ebdd8ac1623defaba8359bce73) (the first commit where all CI checks passed) to one or more additional pull requests? Thanks!

Done, it's still quite a large PR, but I don't see a sensible way to split it further.

File: io/src/openni2_grabber.cpp [hunk 5]
```diff
@@ -633,22 +633,22 @@ pcl::io::OpenNI2Grabber::convertToXYZRGBPointCloud (const Image::Ptr &image, con
   float fx_inv = 1.0f / fx;
   float fy_inv = 1.0f / fy;
 
-  const std::uint16_t* depth_map = (const std::uint16_t*) depth_image->getData ();
+  auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```
Comment (mvieth) [inline io/src/openni2_grabber.cpp:636]: The `const_cast` is strange. How about we use `const auto* depth_map = ...`, apply `fillDepthImageRaw` to `depth_resize_buffer_.data()` and then assign `depth_map = depth_resize_buffer_.data();`? (3 more instances below for rgb_buffer, depth_map, and ir_map)

File: simulation/tools/sim_viewer.cpp [hunk 6]
```diff
@@ -96,7 +96,7 @@ using GeometryHandlerConstPtr = GeometryHandler::ConstPtr;
 #define NORMALS_SCALE 0.01
 #define PC_SCALE 0.001
 
-std::uint16_t t_gamma[2048];
+std::uint16_t t_gamma[2048]{};
```
Comment (mvieth) [inline simulation/tools/sim_viewer.cpp:99]: Why did you add this?

**Review by mvieth — COMMENTED** [review]
Thanks for splitting the PR

File: simulation/tools/sim_viewer.cpp (hunk 6 above)
Comment (gnawme) [reply simulation/tools/sim_viewer.cpp:99]: Zero-initialization. In this case (per Godbolt) it's only a single instruction:

`t_gamma:
  .zero 4096`

Comment (mvieth) [reply simulation/tools/sim_viewer.cpp:99]: But `t_gamma` is filled with the desired values here: https://github.com/PointCloudLibrary/pcl/blob/master/simulation/tools/sim_viewer.cpp#L844 So zero-initialization doesn't seem necessary?

Comment (gnawme) [reply simulation/tools/sim_viewer.cpp]: Fair enough, reverted. The decl is a long way from the point of first use, though...

File: io/src/openni2_grabber.cpp [hunk 7]
```diff
@@ -556,13 +556,13 @@ pcl::io::OpenNI2Grabber::convertToXYZPointCloud (const DepthImage::Ptr& depth_im
 
   float bad_point = std::numeric_limits<float>::quiet_NaN ();
 
-  const std::uint16_t* depth_map = (const std::uint16_t*) depth_image->getData ();
+  auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```
Comment (mvieth) [inline io/src/openni2_grabber.cpp:559]:
```suggestion
  const auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```
Let's keep const here, also for `depth_map` (2x), `rgb_buffer`, and `ir_map` below

**Review by mvieth — APPROVED** [review]
Thanks!
